pytorch_lightning/core/optimizer.py

```diff
@@ -102,11 +102,13 @@ def _on_trainer_init(self, trainer):
                 break
 
     @classmethod
-    def to_lightning_optimizer(cls, optimizer, trainer):
-        if isinstance(optimizer, LightningOptimizer):
-            return optimizer
-        optimizer = cls(optimizer)
-        optimizer._on_trainer_init(trainer)
+    def to_lightning_optimizer(cls, optimizer, opt_idx, trainer):
```

Comment:
```suggestion
    def to_lightning_optimizer(cls, optimizer, trainer, opt_idx):
```
reorder can be breaking API change

Reply:
done